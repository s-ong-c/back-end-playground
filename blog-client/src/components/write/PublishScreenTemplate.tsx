import * as React from 'react';
import styled, { css } from 'styled-components';
import palette from '../../lib/styles/palette';
import zIndexes from '../../lib/styles/zIndexes';
import transitions from '../../lib/styles/transitions';
import HideScroll from '../common/HideScroll';
const PublishScreenTemplateBlock = styled.div<{ visible: boolean }>`
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background: ${palette.gray0};
  z-index: ${zIndexes.PublishScreen};
  animation: ${transitions.slideUp} 0.25s forwards ease-in;
  ${props =>
    props.visible &&
    css`
      animation: ${transitions.slideDown} 0.125s forwards ease-out;
    `}
`;
const Wrapper = styled.div`
  width: 768px;
  display: flex;
`;
const Pane = styled.div`
  flex: 1;
  min-width: 0;
`;
const RightPane = styled.div`
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: space-between;
`;
const Separator = styled.div`
  width: 1px;
  height: 100%;
  min-height: 425px;
  background: ${palette.gray2};
  margin-left: 2rem;
  margin-right: 2rem;
`;

export interface PublishScreenTemplateProps {
  visible: boolean;
  left: React.ReactNode;
  right: React.ReactNode;
}

const { useState, useEffect } = React;
const PublishScreenTemplate: React.SFC<PublishScreenTemplateProps> = ({
  visible,
  left,
  right,
}) => {
  const [animate, setAnimate] = useState(false);

  useEffect(() => {
    let timeoutId: number | null = null;
    if (visible) {
      setAnimate(true);
    } else if (!visible && animate) {
      timeoutId = setTimeout(() => {
        setAnimate(false);
      }, 125);
    }
    return () => {
      if (timeoutId) {
        clearTimeout(timeoutId);
      }
    };
  }, [visible]);
  if (!visible && !animate) return null;
  return (
    <PublishScreenTemplateBlock visible={!visible}>
      <Wrapper>
        <Pane>{left}</Pane>
        <Separator />
        <RightPane>{right}</RightPane>
      </Wrapper>
      {/* <HideScroll /> */}
    </PublishScreenTemplateBlock>
  );
};

export default PublishScreenTemplate;
