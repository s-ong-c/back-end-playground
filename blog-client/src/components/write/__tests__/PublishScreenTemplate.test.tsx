import * as React from 'react';
import { render } from 'react-testing-library';
import PublishScreenTemplate, {
  PublishScreenTemplateProps,
} from '../PublishScreenTemplate';

describe('PublishScreenTemplate', () => {
  const setup = (props: Partial<PublishScreenTemplateProps> = {}) => {
    const initialProps: PublishScreenTemplateProps = {
      visible: true,
      left: 'left',
      right: 'right',
    };
    const utils = render(
      <PublishScreenTemplate {...initialProps} {...props} />,
    );
    return {
      ...utils,
    };
  };
  it('renders property', () => {
    setup();
  });
  it('matches snapshot', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
  it('is not visible when visible is false', () => {
    const utils = setup({ visible: false });
    expect(utils.queryByText('left')).toBeFalsy();
  });
  it('shows left and right', () => {
    const utils = setup();
    utils.getByText('left');
    utils.getByText('right');
  });
});
