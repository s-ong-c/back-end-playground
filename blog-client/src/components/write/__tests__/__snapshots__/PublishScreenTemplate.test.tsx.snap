// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublishScreenTemplate matches snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa jpEzf"
  >
    <div
      class="sc-bwzfXH hVLrlj"
    >
      <div
        class="sc-htpNat iigZIp"
      >
        left
      </div>
      <div
        class="sc-ifAKCX eHlNNW"
      />
      <div
        class="sc-bxivhb kmtPux"
      >
        right
      </div>
    </div>
  </div>
</div>
`;
